include/engine/guidance/route_step.hpp

```diff
@@ -0,0 +1,39 @@
+#ifndef ROUTE_STEP_HPP
+#define ROUTE_STEP_HPP
+
+#include "extractor/travel_mode.hpp"
+#include "engine/guidance/step_maneuver.hpp"
+
+#include <string>
+#include <vector>
+
+namespace osrm
+{
+namespace engine
+{
+namespace guidance
+{
+// Given the following turn from a,b to b,c over b:
+//  a --> b --> c
+// this struct saves the information of the segment b,c.
+// Notable exceptions are Departure and Arrival steps.
+// Departue: s --> a --> b. Represents the segment s,a with location being s.
+// Arrive: a --> b --> t. The segment (b,t) is already covered by the previous segment.
+struct RouteStep
+{
+    unsigned name_id;
+    std::string way_name;
+    double duration;
+    double distance;
```

Comment:
instead of duration and distance, why not use the Route type defined above that itself holds a duration and distance?